x-pack/plugins/monitoring/server/telemetry_collection/get_logstash_stats.ts

```diff
@@ -0,0 +1,423 @@
+  count: number;
+
+  cluster_stats?: {
+    collection_types?: { [collection_type_type: string]: number };
+    queues?: { [queue_type: string]: number };
+    plugins?: Array<{ name: string; count: number }>;
+    pipelines?: {
+      count?: number;
+      batch_size_max?: number;
+      batch_size_avg?: number;
+      batch_size_min?: number;
+      batch_size_total?: number;
+      workers_max?: number;
+      workers_avg?: number;
+      workers_min?: number;
+      workers_total?: number;
+      sources?: { [source_type: string]: boolean };
+    };
+  };
+}
+
+const getLogstashBaseStats = () => ({
+  versions: [],
+  count: 0,
+  cluster_stats: {
+    pipelines: {},
+    plugins: [],
+  },
+});
+
+export interface LogstashStats {
+  cluster_uuid: string;
+  source_node: string;
+  type: string;
+  agent?: {
+    type: string;
+  };
+  logstash_stats?: {
+    pipelines?: [
+      {
+        id?: string;
+        ephemeral_id: string;
+        queue?: {
+          type: string;
+        };
+      }
+    ];
+    logstash?: {
+      version?: string;
+      uuid?: string;
+      snapshot?: string;
+    };
+  };
+}
+
+export interface LogstashState {
+  logstash_state?: {
+    pipeline?: {
+      batch_size?: number;
+      workers?: number;
+      representation?: {
+        graph?: {
+          vertices?: [
+            {
+              config_name?: string;
+              plugin_type?: string;
+              meta?: {
+                source?: {
+                  protocol?: string;
+                };
+              };
+            }
+          ];
+        };
+      };
+    };
+  };
+}
+
+export interface LogstashProcessOptions {
+  clusters: { [clusterUuid: string]: LogstashBaseStats };
+  allEphemeralIds: { [clusterUuid: string]: string[] };
+  versions: { [clusterUuid: string]: Counter };
+  plugins: { [clusterUuid: string]: Counter };
+}
+
+/*
+ * Update a clusters object with processed Logstash stats
+ * @param {Array} results - array of LogstashStats docs from ES
+ * @param {Object} clusters - LogstashBaseStats in an object keyed by the cluster UUIDs
+ * @param {Object} allEphemeralIds - EphemeralIds in an object keyed by cluster UUIDs to track the pipelines for the cluster
+ * @param {Object} versions - Versions in an object keyed by cluster UUIDs to track the logstash versions for the cluster
+ * @param {Object} plugins - plugin information keyed by cluster UUIDs to count the unique plugins
+ */
+export function processStatsResults(
+  results: SearchResponse<LogstashStats>,
+  { clusters, allEphemeralIds, versions, plugins }: LogstashProcessOptions
+) {
+  const currHits = results?.hits?.hits || [];
+  currHits.forEach((hit) => {
+    const clusterUuid = hit._source.cluster_uuid;
+    if (clusters[clusterUuid] === undefined) {
+      clusters[clusterUuid] = getLogstashBaseStats();
+      versions[clusterUuid] = new Map();
+      plugins[clusterUuid] = new Map();
+    }
+
+    const processLogstashStatsResults = () => {
```

Comment:
I wonder why are we defining the method and calling it right after without any params? Is it to have more complete stack trace?